src/controller/CHIPDeviceController.cpp

```diff
@@ -1746,17 +1769,12 @@ void DeviceCommissioner::CommissioningStageComplete(CHIP_ERROR err, Commissionin
 {
     // Once this stage is complete, reset mDeviceBeingCommissioned - this will be reset when the delegate calls the next step.
     MATTER_TRACE_SCOPE("CommissioningStageComplete", "DeviceCommissioner");
-    if (mDeviceBeingCommissioned == nullptr)
-    {
-        // We are getting a stray callback (e.g. due to un-cancellable
-        // operations) when we are not in fact commissioning anything.  Just
-        // ignore it.
-        return;
-    }
+    VerifyOrDie(mDeviceBeingCommissioned);
```

Comment:
Are there no commissioning stages where we are waiting for our API consumer to call back into us?  This seems pretty dangerous to me...

Reply:
`NetworkCredentialsReady` and `ICDRegistrationInfoReady` check `mCommissioningStage` and return an error  if called in the wrong state. `ContinueCommissioningAfterDeviceAttestation` checks both the stage and the device explicitly. Are there any others?

I'm not a huge fan of the kind of defensive programming where internal bugs are expected and ignored, but if we had something like `DebugVerifyOrDieOrReturn()` that could be useful here to only enable these checks in debug builds?

Reply:
Unresolveing this not becuase there's something specific to be done, but just so github doesn't collapse it. Feel free to close if you think there's sufficient debug info exposed.

Reply:
With `CHIP_CONFIG_VERBOSE_VERIFY_OR_DIE` defined (which is true for Darwin and Linux among a few others) it translates to
```
    nlABORT_ACTION(aCondition, ChipLogDetail(Support, "VerifyOrDie failure at %s:%d: %s", __FILE__, __LINE__, #aCondition))
```

It's surprising that this is currently at Detail level, I'll change this to Error.